File: qc-application/src/main/java/com/eager/questioncloud/application/api/feed/library/service/FeedLibraryService.java
package com.eager.questioncloud.application.api.feed.library.service;

import com.eager.questioncloud.core.domain.question.common.QuestionFilter;
import com.eager.questioncloud.core.domain.userquestion.dto.UserQuestionDetail;
import com.eager.questioncloud.core.domain.userquestion.infrastructure.repository.UserQuestionRepository;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class FeedLibraryService {
    private final UserQuestionRepository userQuestionRepository;

    public List<UserQuestionDetail> getUserQuestions(QuestionFilter questionFilter) {
        return userQuestionRepository.getUserQuestions(questionFilter);
    }

    public int countUserQuestions(QuestionFilter questionFilter) {
        return userQuestionRepository.countUserQuestions(questionFilter);
    }
}

File: qc-application/src/main/kotlin/com/eager/questioncloud/application/api/feed/library/service/FeedLibraryService.kt
package com.eager.questioncloud.application.api.feed.library.service

import com.eager.questioncloud.application.api.feed.library.dto.LibraryContent
import com.eager.questioncloud.application.api.feed.library.implement.LibraryContentReader
import com.eager.questioncloud.core.domain.question.common.QuestionFilter
import org.springframework.stereotype.Service

@Service
class FeedLibraryService(
    private val libraryContentReader: LibraryContentReader
) {
    fun getUserQuestions(questionFilter: QuestionFilter): List<LibraryContent> {
        return libraryContentReader.getUserQuestions(questionFilter)
    }

    fun countUserQuestions(questionFilter: QuestionFilter): Int {
        return libraryContentReader.countUserQuestions(questionFilter)
    }
}
